src/onboard_dataflowspec.py

```diff
@@ -221,10 +222,14 @@ def onboard_silver_dataflow_spec(self):
 
         logger.info(silver_transformation_json_file)
 
+        silver_transformation_json_df.show()
+        silver_data_flow_spec_df.show()
+
         silver_data_flow_spec_df = silver_transformation_json_df.join(
             silver_data_flow_spec_df,
-            silver_transformation_json_df.target_table
-            == silver_data_flow_spec_df.targetDetails["table"],
+            (silver_transformation_json_df.target_table == silver_data_flow_spec_df.targetDetails["table"]) &
+            ((silver_transformation_json_df.database.isNull()) | (
+                        silver_transformation_json_df.database == silver_data_flow_spec_df.targetDetails["database"]))
         )
         silver_dataflow_spec_df = (
             silver_data_flow_spec_df.drop("target_table")  # .drop("path")
```

Comment:
need to add unit tests:
1. with new database attribute in silver transformation
2. without database attribute

Reply:
added unit tests:

1. with new database attribute in silver transformation
2. without database attribute
3. not matching value of database attribute (silver transformation vs onboarding template silver table database name)